types/jest/index.d.ts

```diff
@@ -473,6 +493,26 @@ declare namespace jest {
          * to test what arguments it was last called with.
          */
         toHaveBeenLastCalledWith(...params: any[]): R;
+        /**
+         * Ensure that a mock function has returned (as opposed to thrown) at least once.
+         */
+        toHaveReturned(): R;
+        /**
+         * Ensure that a mock function has returned (as opposed to thrown) a specified number of times.
+         */
+        toHaveReturnedTimes(count: number): R;
+        /**
+         * Ensure that a mock function has returned a specified value at least once.
+         */
+        toHaveReturnedWith(value: any): R;
+        /**
+         * Ensure that the last call to a mock function has returned a specified value.
+         */
+        toHaveLastReturnedWith(value: any): R;
+        /**
+         * Ensure that the nth call to a mock function has returned a specified value.
+         */
+        toHaveNthReturnedWith(n: number, value: any): R;
         /**
```

Comment:
I'm stoked for this change to land! Don't forget toStrictEqual()!

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26236/files
https://facebook.github.io/jest/blog/2018/05/29/jest-23-blazing-fast-delightful-testing.html

Reply:
I think we can make another PR for toStrictEqual and toMatchSnapshot 👌since @UselessPickles is only familiar with the matchers he has been using.